### PR TITLE
fix(alarms): Add default alarms to Pocket ALB

### PR DESCRIPTION
**Pull request by kkyeboah**
## Goal

Add default alarms to PocketALBApplication.

The following alarms are added by default:
- HTTP 5xx error rate alarm 
- Latency alarm

### CODE REVIEW

File: src/pocket/PocketALBApplication.ts
Comment (mmiermans) [inline src/pocket/PocketALBApplication.ts:733]:
```suggestion
        alarmDescription: 'Percentage of 5xx responses exceeds threshold'
      },
```

File: src/pocket/PocketALBApplication.ts
Comment (mmiermans) [inline src/pocket/PocketALBApplication.ts:744]:
```suggestion
        alarmDescription: 'Average HTTP response time exceeds threshold',
```
"slower-than-average" implies to me that we'd alarm if the response time falls below the average, instead of simply comparing against a fixed threshold.

File: src/pocket/PocketALBApplication.ts
Comment (mmiermans) [inline src/pocket/PocketALBApplication.ts:760]: Same comment here: I'm assuming the threshold is not the normal, but the maximum threshold allowed.

**Review by mmiermans — COMMENTED** [review]
Code itself looks good (as far as I'm able to review TypeScript without ever having written it :zipper_mouth_face: ). I just have some very minor comments on making the alarm descriptions clearer.

File: src/pocket/PocketALBApplication.ts
Comment (jpetto) [inline src/pocket/PocketALBApplication.ts:150]: i'm guessing the reasoning behind minimizing params on these functions and leveraging a global `config` is less code, but i would argue this makes the functions more brittle and tougher to follow.

i know it's more code to explicitly define what data a particular function needs, but i think doing so makes the function easier to read and understand.

honestly, i think our design here is bad. these functions do too much. my vote here is to revert any config changes and make a ticket to refactor this file into smaller, more specialized functions.

happy to discuss further.

**Review by jpetto — CHANGES_REQUESTED** [review]
i think with the `config` changes here, we're moving in the wrong direction. happy to discuss further.

File: src/pocket/PocketALBApplication.ts
Comment (kkyeboah) [reply src/pocket/PocketALBApplication.ts:150]: I don't agree this makes the functions harder to follow, these are class methods, and I believe it's ok and for me even expected to use class properties within them, especially if they are properties that are accepted through the classes constructor. I agree we should discuss further, possibly in slack?

I'm ok with refactoring into more specialized functions, I can go either way there for this module.

Comment (jpetto) [reply src/pocket/PocketALBApplication.ts:150]: yeah, i think this is worth a discussion. i don't want to block progress on this repo, but i also don't want us to get too far along without assessing our patterns. i'll leave this up to the next reviewer and will approve if we have consensus. i'll also try to write up an issue where we can discuss async.

**Review by jpetto — APPROVED** [review]

**Comment by kkyeboah** [comment]
Discussed with @mmiermans and @jpetto. I will create a proposal for the direction we would like to go with how we create Pocket modules and facilitate a discussion on it.

**Pull request merged**
